src/setup/jest.ts

```diff
@@ -10,40 +15,306 @@ import { setDependencies } from './dependencies'
+  return new Uint8Array(encrypted)
+}
+
+const decrypt = async (encrypted: Uint8Array, keyStr: string): Promise<Uint8Array> => {
+  const key = await aes.importKey(keyStr, { length: SymmKeyLength.B256 })
+  const decryptedBuf = await aes.decryptBytes(encrypted.buffer, key)
+  return new Uint8Array(decryptedBuf)
+}
+
+const genKeyStr = async (): Promise<string> => {
+  const key = await aes.makeKey({ length: SymmKeyLength.B256 })
+  return aes.exportKey(key)
+}
+
+const decryptGCM = async (encrypted: string, keyStr: string, ivStr: string): Promise<string> => {
+  const iv = utils.base64ToArrBuf(ivStr)
+  const sessionKey = await webcrypto.subtle.importKey(
+    "raw",
+    utils.base64ToArrBuf(keyStr),
+    "AES-GCM",
+    false,
+    [ "encrypt", "decrypt" ]
+  )
+
+  // Decrypt secrets
+  const decrypted = await webcrypto.subtle.decrypt(
+    {
+      name: "AES-GCM",
+      iv: iv
+    },
+    sessionKey,
+    utils.base64ToArrBuf(encrypted)
+  )
+  return utils.arrBufToStr(decrypted, CharSize.B8)
+}
+
+const sha256 = async (bytes: Uint8Array): Promise<Uint8Array> => {
   const buf = bytes.buffer
   const hash = await webcrypto.subtle.digest('SHA-256', buf)
   return new Uint8Array(hash)
 }
 
+const rsaVerify = (message: Uint8Array, signature: Uint8Array, publicKey: Uint8Array): Promise<boolean> => {
+  const keyStr = utils.arrBufToBase64(publicKey.buffer)
+  return rsa.verify(message, signature, keyStr)
+}
+
+const ed25519Verify = (message: Uint8Array, signature: Uint8Array, publicKey: Uint8Array): Promise<boolean> => {
+  return ed25519.verify(signature, message, publicKey)
+}
+
+
+
+
+//-------------------------------------
+// Node RSA Keystore
+//-------------------------------------
+
+class InMemoryRSAKeyStore implements KeyStore {
+
+  store: Record<string, any>;
+  cfg: Config;
+
+  constructor(cfg: Config, store: Record<string, any>) {
+    this.cfg = cfg
+    this.store = store
+  }
+
+  static async init(maybeCfg?: Partial<Config>): Promise<InMemoryRSAKeyStore> {
+    const cfg = config.normalize({
+      ...(maybeCfg || {}),
+      type: CryptoSystem.RSA
+    })
+
+    const { rsaSize, hashAlg, readKeyName, writeKeyName } = cfg
+
+    let store: Record<string, any> = {}
+
+    store[readKeyName] = await rsa.makeKeypair(rsaSize, hashAlg, KeyUse.Read);
+    store[writeKeyName] = await rsa.makeKeypair(rsaSize, hashAlg, KeyUse.Write)
+
+    return new InMemoryRSAKeyStore(cfg, store)
+  }
+
+  async readKey() {
+    return this.store[this.cfg.readKeyName]
+  }
+
+  async writeKey() {
+    return this.store[this.cfg.writeKeyName]
+  }
+
+  async getSymmKey(keyName: string, cfg?: Partial<Config>): Promise<CryptoKey> {
+    const mergedCfg = config.merge(this.cfg, cfg)
+    const maybeKey = this.store[keyName]
+    if(maybeKey !== null) {
+      return maybeKey
+    }
+    const key = await aes.makeKey(config.symmKeyOpts(mergedCfg))
+    this.store[keyName] = key
+    return key
+  }
```

Comment:
I duplicate some code here [from keystore-idb](https://github.com/fission-suite/keystore-idb/blob/74d2a96429dde2b86d3daa3597b4ef35334db937/src/rsa/keystore.ts).

We should probably eventually make it possible to use our dependencies in node. However, doing that for keystore-idb could be questionable, given it has "idb" in its name :smile: 

Any way, this feels kind of quick-and-dirty right now. Let me know if you have any other ideas.